cirq/ops/common_gates_test.py

```diff
@@ -272,6 +261,232 @@ def test_h_str():
     assert str(cirq.H**0.5) == 'H^0.5'
 
 
+def test_x_act_on():
```

Comment:
Create a testing method `def assert_clifford_tableau_effect_matches_unitary():`

Implement the method as follows:

1. Create a circuit that prepares a Bell pair for each qubit to be input into the gate (H + CNOT), then runs the gate (acting on one qubit from each Bell pair).
2. Create a method `stabilizers_to_state_vector` which creates the state stabilized by a given list of commuting stabilizers. Do this by generating a random state and then iteratively projecting it onto each stabilizer's +1 eigenstate.
3. Compare the result of `cirq.final_state_vector(circuit)` to (the equivalent of) `stabilizers_to_state(cirq.final_clifford_tableau(circuit))`.

Alternatively, you can check that the state returned by the state vector simulator is in fact stabilized by each of the claimed stabilizers in the Clifford tableau. That would be more efficient in general.

The key thing is that we want to be able to base the correctness of the Clifford simulation on the correctness of the state vector simulator.

Reply:
Maybe we could implement a utility method `state_vector_has_stabilizer(np.ndarray, PauliString)`.

Reply:
Sounds good. Went with the state_vector_has_stabilizer approach.